zombies.js

```diff
@@ -79,135 +104,224 @@
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+
+  this.discardItem = (item) => {
+    let checkBagForItem = this.getPack().indexOf(item);
+    if (checkBagForItem !== -1) {
+      this.getPack().splice(checkBagForItem, 1);
+      console.log(this.name + ' discarded ' + item.name);
+      return true;
+    }
+    console.log('Item not in bag');
+    return false;
+  }
+
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  this.equip = (itemToEquip) => {
+    let checkBagForWeapon = this.getPack().indexOf(itemToEquip);
+    if ((itemToEquip instanceof Weapon) && checkBagForWeapon !== -1) {
+      if (this.equipped === false) {
+        this.getPack().splice(checkBagForWeapon, 1);
+        this.equipped = itemToEquip;
+        this.discardItem(itemToEquip);
+      } else {
+        this.getPack().splice(checkBagForWeapon, 1, this.equipped);
+        this.equipped = itemToEquip;
+        console.log(this.name + ' has equipped ' + itemToEquip.name);
+      }
+    }
+  }
+
+
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  this.eat = (itemToEat) => {
+    let checkBagForFood = this.getPack().indexOf(itemToEat);
+    if (itemToEat instanceof Food && checkBagForFood > -1) {
+      this.getPack().splice(checkBagForFood, 1);
+      this.health += itemToEat.energy;
+      if (this.health + itemToEat.energy > this.getMaxHealth()) {
```

Comment:
This if conditional actually adds the food energy to the Player's health twice now.